src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/EventCounter/EventCounterTriggerSettings.cs

```diff
@@ -0,0 +1,95 @@
+        IValidatableObject
+    {
+        internal const int CounterIntervalSeconds_MaxValue = 24 * 60 * 60; // 1 day
+        internal const int CounterIntervalSeconds_MinValue = 1; // 1 second
+
+        internal const string EitherGreaterThanLessThanMessage = "Either the " + nameof(GreaterThan) + " field or the " + nameof(LessThan) + " field are required.";
+
+        internal const string GreaterThanMustBeLessThanLessThanMessage = "The " + nameof(GreaterThan) + " field must be less than the " + nameof(LessThan) + " field.";
+
+        internal const string SlidingWindowDuration_MaxValue = "1.00:00:00"; // 1 day
+        internal const string SlidingWindowDuration_MinValue = "00:00:01"; // 1 second
+
+        /// <summary>
+        /// The name of the event provider from which counters will be monitored.
+        /// </summary>
+        [Required]
+        public string ProviderName { get; set; }
+
+        /// <summary>
+        /// The name of the event counter from the event provider to monitor.
+        /// </summary>
+        [Required]
+        public string CounterName { get; set; }
+
+        /// <summary>
+        /// The lower bound threshold that the event counter value must hold for
+        /// the duration specified in <see cref="SlidingWindowDuration"/>.
+        /// </summary>
+        public double? GreaterThan { get; set; }
+
+        /// <summary>
+        /// The upper bound threshold that the event counter value must hold for
+        /// the duration specified in <see cref="SlidingWindowDuration"/>.
+        /// </summary>
+        public double? LessThan { get; set; }
+
+        /// <summary>
+        /// The sliding duration of time in which the event counter must maintain a value
+        /// above, below, or between the thresholds specified by <see cref="GreaterThan"/> and <see cref="LessThan"/>.
+        /// </summary>
+        [Range(typeof(TimeSpan), SlidingWindowDuration_MinValue, SlidingWindowDuration_MaxValue)]
+        public TimeSpan SlidingWindowDuration { get; set; }
+
+        /// <summary>
+        /// The sampling interval of the event counter.
+        /// </summary>
+        [Range(CounterIntervalSeconds_MinValue, CounterIntervalSeconds_MaxValue)]
+        public int CounterIntervalSeconds { get; set; }
+
+        IEnumerable<ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
+        {
+            List<ValidationResult> results = new();
+
+            if (!GreaterThan.HasValue && !LessThan.HasValue)
+            {
+                results.Add(new ValidationResult(
+                    EitherGreaterThanLessThanMessage,
+                    new[]
+                    {
+                        nameof(GreaterThan),
+                        nameof(LessThan)
+                    }));
+            }
+            else if (GreaterThan.HasValue && LessThan.HasValue)
+            {
+                if (GreaterThan.Value > LessThan.Value)
```

Comment:
Should this technically be >= instead? Based off the message on line 22, we imply that GreaterThan < LessThan; thus, we would want to fail when GreaterThan >= LessThan